File: herencia-dto/HerenciaDto-Java-IvanRC/src/main/java/storage/personas/PersonaStorageServiceXml.java
package storage.personas;

import config.ConfigApp;
import dto.ListaPersonasDto;
import mapper.PersonaMapper;
import models.Persona;
import org.simpleframework.xml.core.Persister;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class PersonaStorageServiceXml implements PersonasStorageService{

    ConfigApp configApp = ConfigApp.getInstance();
    File file = new File(configApp.getAPP_DATA()+File.separator+"personas.xml");

    Persister persister = new Persister();

    public PersonaStorageServiceXml() throws IOException {}

    @Override
    public void saveAll(List<Persona> entities) {
        try {
            persister.write(PersonaMapper.toPersonasDto(entities), file);
        }catch (Exception e){
            System.err.println(e.getMessage());
        }
    }

    @Override
    public List<Persona> loadAll() throws FileNotFoundException {
        List<Persona> personas = new ArrayList<>();
        if(!file.exists()) return personas;
        try {
            personas = PersonaMapper.toPersonas(persister.read(ListaPersonasDto.class, file));
        }catch (Exception e){
            System.err.println(e.getMessage());
        }
        return personas;
    }
}

File: herencia-dto/HerenciaDto-Kotlin-IvanRC/src/main/kotlin/storage/persona/PersonaStorageServiceXml.kt
package storage.persona

import config.ConfigApp
import dto.ListaPersonasDto
import mapper.toPersonas
import mapper.toPersonasDto
import models.Persona
import mu.KotlinLogging
import org.simpleframework.xml.core.Persister
import java.io.File

class PersonaStorageServiceXml: PersonaStorageService {

    private val configApp = ConfigApp
    private val file = File(configApp.APP_DATA+ File.separator+"personas.xml")
    private val persister = Persister()

    private val logger = KotlinLogging.logger {  }

    override fun saveAll(entities: List<Persona>) {
        logger.debug { "Se guardan todas las personas en el fichero XML" }
        persister.write(entities.toPersonasDto(), file)
    }

    override fun loadAll(): List<Persona> {
        logger.debug { "Se cargan todas las personas del fichero XML" }
        if(!file.exists()) return emptyList()
        return persister.read(ListaPersonasDto::class.java, file).toPersonas()
    }
}